### PR TITLE
qinfeng/delta_surface_market_convention_fix

### PR DESCRIPTION
**Pull request by qinfeng-deriv**
https://redmine.deriv.cloud/issues/85026

### CODE REVIEW

File: lib/VolSurface/Utils.pm [hunk 1]
```diff
@@ -89,16 +89,20 @@ sub get_delta_for_strike {
         croak "Arg $_ is undef at get_delta_for_strike" unless defined $args->{$_};
     }
 
+    $new_args{forward} = $new_args{t} >= 1 ? 1 : 0;
```
Comment (VishalMenon-Deriv) [inline lib/VolSurface/Utils.pm]: We can get rid of this line as we already included it as a parameter in the function calls of [Quant/Framework/VolSurface/Delta.pm](https://github.com/regentmarkets/perl-Quant-Framework/blob/afbbbf86e7db23354692e82f62349725e4346e0c/lib/Quant/Framework/VolSurface/Delta.pm)

File: lib/VolSurface/Utils.pm [hunk 2]
```diff
@@ -89,16 +89,20 @@ sub get_delta_for_strike {
...
+
     my ($K, $sigma, $t, $S, $r, $q, $premium_adjusted) =
         ($new_args{strike}, $new_args{atm_vol}, $new_args{t}, $new_args{spot}, $new_args{r_rate}, $new_args{q_rate}, $new_args{premium_adjusted});
 
     my $delta;
     if ($premium_adjusted) {
+        my $forward_adj = $new_args{forward} ? 1 : exp(-1 * $new_args{r_rate} * $new_args{t});
```
Comment (VishalMenon-Deriv) [inline lib/VolSurface/Utils.pm]: Turns out this is actually supposed to be `$new_args{forward} ? exp($new_args{q_rate} * $new_args{t}) : 1`. Note the lack of negative next to `q_rate`

File: lib/VolSurface/Utils.pm [hunk 3]
```diff
@@ -89,16 +89,20 @@ sub get_delta_for_strike {
...
         my $d2 = (log($S / $K) + ($r - $q - ($sigma**2) / 2) * $t) / ($sigma * sqrt($t));
-        $delta = ($K / $S) * exp(-1 * $r * $t) * pnorm($d2);
+        $delta = ($K / $S) * $forward_adj * pnorm($d2);
```
Comment (VishalMenon-Deriv) [inline lib/VolSurface/Utils.pm]: And here is where the `r_rate` actually comes in. This line will be 
```perl
$delta = ($K / $S) * $forward_adj * exp(-1 * $new_args{r_rate} * $new_args{t}) * pnorm($d2);
```

File: lib/VolSurface/Utils.pm [hunk 4]
```diff
@@ -140,6 +144,14 @@ sub get_strike_for_spot_delta {
         croak 'Provided delta [' . $new_args{delta} . '] must be on [0,1]';
     }
 
+    if ($new_args{forward}) {
```
Comment (VishalMenon-Deriv) [inline lib/VolSurface/Utils.pm:147]: As per the previous comments, it turns out that this entire if statement can be (re)condensed into

```perl
if ($new_args{forward}) {
    $new_args{delta} = $new_args{delta} * exp(-$new_args{q_rate} * $new_args{t});
}
```

Basically just the else condition.

File: lib/VolSurface/Utils.pm [hunk 5]
```diff
@@ -140,6 +144,14 @@ sub get_strike_for_spot_delta {
...
+        if ($new_args{premium_adjusted}){
+            $new_args{delta} = $new_args{delta} * exp(-$new_args{r_rate} * $new_args{t});
+        } else {
+            $new_args{delta} = $new_args{delta} * exp(-$new_args{q_rate} * $new_args{t});
+        }
+    }
+
```
Comment (VishalMenon-Deriv) [inline lib/VolSurface/Utils.pm]: As per the previous comments, it turns out that this entire if statement can be (re)condensed into

```perl
if ($new_args{forward}) {
    $new_args{delta} = $new_args{delta} * exp(-$new_args{q_rate} * $new_args{t});
}
```

Basically just the else condition.

**Review by VishalMenon-Deriv — COMMENTED** [review]
Modifications to how deltas are calculated, as according to https://quantpie.co.uk/fx/fx_delta.php.

File: README.md [hunk 6]
```diff
@@ -10,7 +10,7 @@ A class that handles several volatility related methods such as gets strikes fro
 
 use VolSurface::Utils;
 
-my $delta = get_delta_for_strike({ strike => $strike, atm_vol => $atm_vol, t => $t, spot => $spot, r_rate =>$r_rate, q_rate => $q_rate, premium_adjusted => $premium_adjusted });
+my $delta = get_delta_for_strike({ strike => $strike, atm_vol => $atm_vol, t => $t, spot => $spot, r_rate =>$r_rate, q_rate => $q_rate, premium_adjusted => $premium_adjusted , $forward => forward});
```
Comment (VishalMenon-Deriv) [inline README.md]: I think this is meant to be `forward => $forward`

File: README.md [hunk 7]
```diff
@@ -10,7 +10,7 @@ A class that handles several volatility related methods such as gets strikes fro
 
 use VolSurface::Utils;
 
-my $delta = get_delta_for_strike({ strike => $strike, atm_vol => $atm_vol, t => $t, spot => $spot, r_rate =>$r_rate, q_rate => $q_rate, premium_adjusted => $premium_adjusted });
+my $delta = get_delta_for_strike({ strike => $strike, atm_vol => $atm_vol, t => $t, spot => $spot, r_rate =>$r_rate, q_rate => $q_rate, premium_adjusted => $premium_adjusted , $forward => $forward});
```
Comment (chylli-deriv) [inline README.md]:
```suggestion
my $delta = get_delta_for_strike({ strike => $strike, atm_vol => $atm_vol, t => $t, spot => $spot, r_rate =>$r_rate, q_rate => $q_rate, premium_adjusted => $premium_adjusted , forward => $forward});
```

**Comment by qinfeng-deriv** [comment]
Pulling from @chylli-deriv 's branch to get ci running

**Pull request merged**